packages/gatsby/src/utils/webpack.config.js

```diff
@@ -370,6 +370,10 @@ module.exports = async (
       ],
       alias: {
         gatsby$: directoryPath(path.join(`.cache`, `gatsby-browser-entry.js`)),
+        'core-js': path.dirname(require.resolve(`core-js/package.json`)),
```

Comment:
don't you want to resolve just `core-js` not the package.json file?

Reply:
If I wanted to resolve `core-js`, it would return the file inside the `package.json` `main` field of `core-js`, which is `core-js/index.js`. If I resolved it like that, Gatsby would throw several exceptions when trying to access `[some-path]/node_modules/core-js/index.js/../[some-path]`.

Reply:
well... returning the `package.json` isn't what people are expecting when you require a core-js module ;-)

I'm not super familiar with webpack's aliasing — perhaps this will help you https://webpack.js.org/configuration/resolve/

Reply:
I'm not returning the `package.json`, but the directory of the module. (See the `path.dirname` function call)

Reply:
Further explanation can be found at https://stackoverflow.com/a/49455609/6420957

Reply:
Oh strange. Wasn't aware of that pattern. Could you add a comment there then explaining why the `package.json` is necessary and link to the stackoverflow answer?

Reply:
Sure, I tried to briefly explain the situation.

Reply:
might be clearer to do `path.resolve('../node_modules/core-js')`